dwave/plugins/torch/models/boltzmann_machine.py

```diff
@@ -662,3 +662,277 @@ def estimate_beta(self, spins: torch.Tensor) -> float:
         return beta
+
+class RestrictedBoltzmannMachine(torch.nn.Module):
+    """A Restricted Boltzmann Machine (RBM) model.
+
+    This class defines the parameterization and inference of a binary RBM.
+    Training is performed using Persistent Contrastive Divergence (PCD).
+
+    Args:
+        n_visible (int): Number of visible units.
+        n_hidden (int): Number of hidden units.
+    """
+
+    def __init__(
+        self,
+        n_visible: int,
+        n_hidden: int,
+    ) -> None:
+        super().__init__()
+
+        # Model hyperparameters
+        self._n_visible = n_visible
+        self._n_hidden = n_hidden
+
+        # Initialize model parameters      
+        # initialize weights
+        self._weights = torch.nn.Parameter(
+            0.1 * torch.randn(n_visible, n_hidden)
+        )
+        # initialize visible units biases.
+        self._visible_biases = torch.nn.Parameter(
+            0.5 * torch.ones(n_visible)
+        )  
+        # initialize hidden units biases.
+        self._hidden_biases = torch.nn.Parameter(
+            0.5 * torch.ones(n_hidden)
+        )
+
+        # Stores the last visible states to initialize the Markov chain in Persistent Contrastive Divergence (PCD)
+        self.register_buffer("_previous_visible_values", None)
+
+        # Initialize momenta tensors for momentum-based updates (all start at 0)
+        self.register_buffer("_weight_momenta", torch.zeros(n_visible, n_hidden))
+        self.register_buffer("_visible_bias_momenta", torch.zeros(n_visible))
+        self.register_buffer("_hidden_bias_momenta", torch.zeros(n_hidden))
+
+    @property
+    def n_visible(self) -> int:
+        """Number of visible units."""
+        return self._n_visible
+
+    @property
+    def n_hidden(self) -> int:
+        """Number of hidden units."""
+        return self._n_hidden
+    
+    @property
+    def weights(self) -> torch.Tensor:
+        """Weights of the RBM."""
+        return self._weights
+    
+    @property
+    def visible_biases(self) -> torch.Tensor:
+        """Visible biases of the RBM."""
+        return self._visible_biases
+    
+    @property
+    def hidden_biases(self) -> torch.Tensor:
+        """Hidden biases of the RBM."""
+        return self._hidden_biases
+    
+    @property
+    def previous_visible_values(self) -> torch.Tensor:
+        """Previous visible values used in Persistent Contrastive Divergence (PCD)."""
+        return self._previous_visible_values
+    
+    @property
+    def weight_momenta(self) -> torch.Tensor:
+        """Weight momenta of the RBM."""
+        return self._weight_momenta
+    
+    @property
+    def visible_bias_momenta(self) -> torch.Tensor:
+        """Visible bias momenta of the RBM."""
+        return self._visible_bias_momenta
+    
+    @property
+    def hidden_bias_momenta(self) -> torch.Tensor:
+        """Hidden bias momenta of the RBM."""
+        return self._hidden_bias_momenta
```

Comment:
Are all of these useful attributes to access for a user? Consider removing some of these properties if they're only used within the class and not useful for a general user.